cocos/base/CCConsole.cpp

```diff
@@ -937,17 +939,26 @@ bool Console::parseCommand(int fd)
         }
     }
     std::string cmdLine;
-    
-    std::vector<std::string> args;
     cmdLine = std::string(buf);
+    auto commands = Console::Utility::split(cmdLine, _commandSeparator);
+    try {
+        for(auto command : commands) {
+            performCommand(fd, Console::Utility::trim(command));
+        }
+    } catch (const std::runtime_error& e) {
+        Console::Utility::sendToConsole(fd, e.what(), strlen(e.what()));
+    }
     
-    args = Console::Utility::split(cmdLine, ' ');
-    if(args.empty())
-    {
-        const char err[] = "Unknown command. Type 'help' for options\n";
-        Console::Utility::sendToConsole(fd, err, strlen(err));
-        Console::Utility::sendPrompt(fd);
-        return true;
+    
+    Console::Utility::sendPrompt(fd);
+    
+    return true;
+}
+
+void Console::performCommand(int fd, const std::string& command) {
+    std::vector<std::string> args = Console::Utility::split(command, ' ');
+    if(args.empty()) {
+        throw std::runtime_error("Unknown command. Type 'help' for options\n");
```

Comment:
Can we use  return value instead of exception?

Reply:
Yes, if this conflicts with principles to use of exceptions in the code. Be that as it may, I plan to rewrite all the code in Console-subsystem. As practice has shown, this functionality is the most important for identifying problems. The description of problems (Steps To Reproduce) can be completely replaced with a list of class commands. Especially if you implement the command "`sleep <ms>`". For example `<do something> | sleep 5ms | <do something>`. Difficult identifiable problems associated with (for example: Actions or Scene Transitions) can now easily be reproduced without fear that the network problems will affect.